test/stdlib/StringCompatibilityDiagnostics3.swift

```diff
@@ -19,7 +19,8 @@ func testPopFirst() {
   var charSubView: Substring.CharacterView // expected-warning{{'CharacterView' is deprecated: Please use String or Substring directly}}
   charSubView = substr.characters // expected-warning{{'characters' is deprecated: Please use String or Substring directly}}
   dump(charSubView)
-}
-
 
+  var _ = String(str.utf8) ?? "" // expected-warning{{'init' is deprecated: Failable initializer was removed in Swift 4. When upgrading to Swift 4, please use non-failable String.init(_:UTF8View)}}
+  var _: String = String(str.utf8) // expected-error{{'init' is unavailable: Please use failable String.init?(_:UTF8View) when in Swift 3.2 mode}}
```

Comment:
It did not exist in Swift 3, so there is no existing compile-able code that would use this construct, but a nice QoI touch, I guess. :+1:

Reply:
In this specific case, I think it's needed. If people take the advice of the other message to try to use the non-failable one, we give a message back to them that they should in fact still use the failable one.